### DISCUSSION TITLE
Private key handling outside of BIP44 or BIP32 derivation

### DISCUSSION THREAD

**Discussion by shuse2** [MetaMask/snaps#2134] [Q&A]
Hello,
I have question regarding handling the private key.

I have different derivation protocol than BIP44 or BIP32 need to support.
To be precise, `sign.keyPair.fromSeed(hash(recovery phrase))` using `ed25519`. 

Is it possible to access the imported root private key from snap?
Otherwise, only option I see is that importing recovery passphrase or private key from UI to snap directly through `wallet_invokeSnap`.

Security wise is it allowed or safe to communicate the private key or secret passphrase between dapp and snap? (Only once when importing)

**Comment by Montoya** [comment]
Hi @shuse2, thanks for reaching out. Could you tell me which protocol you are trying to support with this derivation?

**Reply by shuse2** [reply]
Im trying to integrate Lisk using the snap.
The derivation is specific to the legacy protocol on the previous version that I want to support
